driver-legacy/src/main/com/mongodb/MongoClientOptions.java

```diff
@@ -142,7 +144,8 @@ private MongoClientOptions(final Builder builder) {
                 .maxSize(getConnectionsPerHost())
                 .maxWaitTime(getMaxWaitTime(), MILLISECONDS)
                 .maxConnectionIdleTime(getMaxConnectionIdleTime(), MILLISECONDS)
-                .maxConnectionLifeTime(getMaxConnectionLifeTime(), MILLISECONDS);
+                .maxConnectionLifeTime(getMaxConnectionLifeTime(), MILLISECONDS)
+                .maxConnecting(builder.maxConnecting);
```

Comment:
Why is this the only one that comes from the builder rather than a getter?  Does it need to be different?

Reply:
While it seems off, the right question is "why does the `MongoClientOptions` duplicates `maxConnectionLifeTime` and other fields from `MongoClientOptions.connectionPoolSettings`?" As a result of duplicating them, the constructor of `MongoClientOptions` at first initializes the duplicated fields from the `builder`, and then initializes the fields in its `connectionPoolSettings` by reading the duplicated fields via methods like `MongoClientOptions.getMaxConnectionLifeTime`.

Instead of doing the same duplication for `maxConnecting`, I use `connectionPoolSettings.maxConnecting`.

Reply:
Maybe a philosophical difference.  I find it confusing as a reader when two styles are mixed in the same class like this. I start wondering if there is a functional difference between the two approaches, and it wastes time.  But if you feel strongly about it, I won't insist. Perhaps a comment explaining the difference would be helpful for future readers.

Reply:
Duplicated `maxConnecting` in `MongoClientOptions` similarly to other fields and filed [Refactor `MongoClientOptions` so that it wraps `MongoClientSettings`](https://jira.mongodb.org/browse/JAVA-4404) as you suggested in Slack.